Userland/Libraries/LibGfx/ImageFormats/BilevelImage.h

```diff
@@ -81,15 +84,41 @@ class BilevelImage : public RefCounted<BilevelImage> {
     Bytes bytes() { return m_bits.bytes(); }
 
 private:
-    BilevelImage(ByteBuffer, size_t width, size_t height, size_t pitch);
+    template<OneOf<BilevelImage, BilevelSubImage> InputType, BilevelImage::CompositionType operator_>
+    friend void composite_onto(InputType const& in, BilevelImage& out, IntPoint position);
 
-    template<CompositionType>
-    void composite_onto(BilevelImage& out, IntPoint position) const;
+    BilevelImage(ByteBuffer, size_t width, size_t height, size_t pitch);
 
     ByteBuffer m_bits;
     size_t m_width { 0 };
     size_t m_height { 0 };
     size_t m_pitch { 0 };
 };
 
+class BilevelSubImage {
```

Comment:
bikeshed: I wonder if we should call this BilevelImageView, or BilevelImageSpan, to be consistent with e.g. AK/Bitmap.h and AK/BitmapView.h.

There's also JPEG2000Span2D.h, which is somewhat similar.

(No action needed.)